truffle/src/main/ruby/core/process_mirror.rb

```diff
@@ -267,6 +269,68 @@ def convert_env_value(value)
           "a+" => ::File::RDWR   | ::File::APPEND | ::File::CREAT
         }
 
+        def spawn_setup
+          require 'fcntl'
+
+          env = options&.delete(:unsetenv_others) ? {} : ENV.to_hash
+          if add_to_env = options&.delete(:env)
+            env.merge! Hash[add_to_env]
+          end
+
+          @env_array = env.map { |k, v| "#{k}=#{v}" }
+
+          if @options
+            pgroup = options[:pgroup]
+            if pgroup
+              Truffle::POSIX.setpgid(0, pgroup)
+            end
```

Comment:
Somehow I adapted and don't recall having an unintentional `=` in a `if` for ages. I think it's not the worse mistake and relatively easy to catch.